api/src/main/java/com/spotify/ffwd/output/OutputPlugin.java

```diff
@@ -23,24 +23,31 @@
 import com.google.inject.PrivateModule;
 import com.google.inject.name.Names;
 import com.spotify.ffwd.filter.Filter;
+import com.spotify.ffwd.module.Batching;
 import java.util.Optional;
 
 @JsonTypeInfo(use = Id.NAME, include = As.PROPERTY, property = "type")
 public abstract class OutputPlugin {
 
-    protected final Optional<Long> flushInterval;
+    protected final Batching batching;
     protected final Optional<Filter> filter;
 
     public OutputPlugin() {
         filter = Optional.empty();
-        flushInterval = Optional.empty();
+        batching = Batching.empty();
     }
 
     public OutputPlugin(
-        final Optional<Filter> filter, final Optional<Long> flushInterval
+        final Optional<Filter> filter, final Batching batching
     ) {
         this.filter = filter;
-        this.flushInterval = flushInterval;
+        this.batching = batching;
+        if (filter == null) {
```

Comment:
filter could be optional in configuration

Reply:
I don't think so. It's an optional, so it will be Optional.empty() if it wasn't specified in the configuration. It will not be null.
I actually remember thinking previously that this was unnecessary, so thank you, I was reminded now :) I've removed this check now.